src/filters/index.ts

```diff
@@ -0,0 +1,57 @@
+import type { Repo } from "~common/repo";
+import NpmFilter from "./npm";
+
+const filters: { [key: string]: RepoFilter } = {
+  npm: NpmFilter,
+}
+
+const defaultFilter = (repo: Repo) => {
+  return []
+}
+
+type FilterFn = typeof defaultFilter
+
+export type RepoFilter = {
+  name: string;
+  searchTerms: string[];
+  fn: FilterFn;
+}
+
+export type FilterMatchHeuristic = {
+  name: string;
+  dataPoints: any[];
+}
+
+const prepareSearch = (repo: Repo) => {
+  for (const filterName in filters) {
```

Comment:
I think here if you did `of` you could actually get the object rather than just the key

so 

```
for (const filter of filters)
```

would actually give you the filter and then you don't have to do the

`const filter = filters[filterName];` on line 28